crates/egui/src/widgets/plot/mod.rs

```diff
@@ -651,10 +737,148 @@ impl Plot {
                 .at_least(min_size.y);
             vec2(width, height)
         };
+        // Determine complete rect of widget.
+        let complete_rect = Rect {
+            min: pos,
+            max: pos + size,
+        };
+        // Next we want to create this layout.
+        // Incides are only examples.
+        //
+        //  +-b-+---------x----------+   +
+        //  |   |      x-Axis 3      |
+        //  c   +--------------------+
+        //  |   |      x-Axis 2      |
+        //  +-+-+--------------------+-+-+
+        //  |y|y|                    |y|y|
+        //  |-|-|                    |-|-|
+        //  |A|A|                    |A|A|
+        // y|x|x|    Plot Window     |x|x|
+        //  |i|i|                    |i|i|
+        //  |s|s|                    |s|s|
+        //  |1|0|                    |2|3|
+        //  +-+-+--------------------+-+-+
+        //      |      x-Axis 0      |   |
+        //      +--------------------+   a
+        //      |      x-Axis 1      |   |
+        //  +   +--------------------+-d-+
+        //
+
+        let mut plot_rect: Rect = {
+            // find dimensions of axis labels
+            // for a, b, c, d meanings see picture
+            let mut a = 0.0;
+            let mut b = 0.0;
+            let mut c = 0.0;
+            let mut d = 0.0;
+            if show_axes.x {
```

Comment:
isn't it a bit superfluous with the `show_axes` setting? Why would the user add axis if they don't want to show them?